staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -356,5 +357,7 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 	if a.onRotate != nil && oldCreds != nil && !reflect.DeepEqual(oldCreds.cert, a.cachedCreds.cert) {
 		a.onRotate()
 	}
+	metrics.ClientCertRotationAge.Observe(oldCreds.cert.Leaf.NotAfter.Sub(oldCreds.cert.Leaf.NotBefore))
+	metrics.ClientCertExpiration.Set(newCreds.cert.Leaf.NotAfter)
```

Comment:
does this metric assume that there's only a single client reporting this metric in the process? if there were two, couldn't this ping-pong between two values?

Reply:
only call the client cert metrics if newCred.cert is non-nil

Reply:
fixed